File: src/tipos/estudiante_pgcrp.ts
export interface EstudiantePgcrp {
  id_estudiante: number;
  cedula: number;
  nombres: string;
  apellidos: string;
  id_extra_catedra?: number;
  actividad_pgcrp?: string;
  id_periodo: number;
  observaciones?: string;
  fecha_asignacion?: string;
}

export interface AsignacionEstudiantePgcrp {
  id_estudiante: number;
  id_extra_catedra?: number;
  id_periodo: number;
  observaciones?: string;
}

export interface EstudiantePgcrpDetalle {
  id_estudiante: number;
  cedula: number;
  nombres: string;
  apellidos: string;
  nombre_grado: string;
  nombre_seccion: string;
  nombre_modalidad: string;
  id_extra_catedra?: number;
  actividad_pgcrp?: string;
  actividad_seccion?: string;
  observaciones?: string;
  fecha_asignacion?: string;
}

export interface ActividadPgcrp {
  id_extra_catedra: number;
  nombre: string;
}

export interface EstadisticasPgcrpSeccion {
  totalEstudiantes: number;
  conPgcrpIndividual: number;
  conPgcrpSeccion: number;
  sinAsignar: number;
  porcentajeAsignados: number;
}

// Tipos para filtros y búsquedas
export interface FiltroEstudiantePgcrp {
  texto?: string;
  tipoAsignacion?: 'todos' | 'individual' | 'seccion' | 'sin_asignar';
  actividad?: number;
}

// Tipos para reportes
export interface ReporteEstudiantePgcrp {
  seccion: string;
  estudiante: string;
  cedula: number;
  pgcrpActual: string;
  tipoAsignacion: 'Individual' | 'Por Sección' | 'Sin Asignar';
  observaciones?: string;
  fechaAsignacion?: string;
}

// Tipo para el estado de una asignación PGCRP
export interface EstadoPgcrp {
  tipo: 'Individual' | 'Por Sección' | 'Sin Asignar';
  actividad: string;
  color: string;
} 

File: src-tauri/src/models/estudiante_pgcrp.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct EstudiantePgcrp {
    pub id_estudiante: i32,
    pub cedula: i64,
    pub nombres: String,
    pub apellidos: String,
    pub id_extra_catedra: Option<i32>,
    pub actividad_pgcrp: Option<String>,
    pub id_periodo: i32,
    pub observaciones: Option<String>,
    pub fecha_asignacion: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct AsignacionEstudiantePgcrp {
    pub id_estudiante: i32,
    pub id_extra_catedra: Option<i32>,
    pub id_periodo: i32,
    pub observaciones: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct EstudiantePgcrpDetalle {
    pub id_estudiante: i32,
    pub cedula: i64,
    pub nombres: String,
    pub apellidos: String,
    pub nombre_grado: String,
    pub nombre_seccion: String,
    pub nombre_modalidad: String,
    pub id_extra_catedra: Option<i32>, // ID de la actividad PGCRP individual
    pub actividad_pgcrp: Option<String>,
    pub actividad_seccion: Option<String>, // PGCRP asignado por sección
    pub observaciones: Option<String>,
    pub fecha_asignacion: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ActividadPgcrp {
    pub id_extra_catedra: i32,
    pub nombre: String,
} 